### PR TITLE
samples: drivers: watchdog: Update sample to use alias

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Simplify sample by using alias for node access. Also removes
usage of DT_LABEL.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

File: samples/drivers/watchdog/src/main.c [hunk 1]
```diff
@@ -14,39 +14,12 @@
...
-/*
- * To use this sample, either the devicetree's /aliases must have a
- * 'watchdog0' property, or one of the following watchdog compatibles
- * must have an enabled node.
- */
-#if DT_NODE_HAS_STATUS(DT_ALIAS(watchdog0), okay)
-#define WDT_NODE DT_ALIAS(watchdog0)
-#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_window_watchdog)
-#define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
-#define WDT_MAX_WINDOW  100U
```
Comment (martinjaeger) [inline samples/drivers/watchdog/src/main.c:26]: This special case still needs to be considered.

**Review by martinjaeger — COMMENTED** [review]
With this change many boards won't work for this sample anymore as they don't have the alias defined. I suggest to add the alias to all boards which have `status = "okay"` for their watchdog in their board `.dts` file and not only in the overlays.

File: samples/drivers/watchdog/src/main.c (hunk 1 above)
Comment (benjaminbjornsson) [reply samples/drivers/watchdog/src/main.c:26]: Oh, I didn’t see that one.. thanks!

**Comment by benjaminbjornsson** [comment]
> With this change many boards won't work for this sample anymore as they don't have the alias defined. I suggest to add the alias to all boards which have `status = "okay"` for their watchdog in their board `.dts` file and not only in the overlays.

Great suggestion, will update!

File: samples/drivers/watchdog/src/main.c (hunk 1 above)
Comment (gmarull) [reply samples/drivers/watchdog/src/main.c:26]: this is a sympthom that we likely need a watchdog specifier, like we have e.g., for adc or pwm.

**Review by gmarull — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]
Thanks for the update, LGTM.

**Review by MaureenHelm — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**